File: 9.20/oc网络封装工具类/oc网络封装工具类/NetworkTool.m
//
//  NetworkTool.m
//  oc网络封装工具类
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//

#import "NetworkTool.h"


@implementation NetworkTool
+(instancetype)sharedtool{
    static NetworkTool *tool;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{

        tool = [[NetworkTool alloc] init];
//        
        NSURL *url = [NSURL URLWithString:@"http://www.weather.com.cn/data/"];
        tool = [[NetworkTool alloc] initWithBaseURL:url];
        
        //增加数据接收类型/sk/101010100.html
        tool.responseSerializer.acceptableContentTypes = [NSSet setWithObjects:@"application/json", @"text/json", @"text/javascript", @"text/html", @"text/plain", nil];
    
    });
    return  tool;
}

-(void)requestType:(RequsetType)type url:(NSString *)url parameters:(id)parameters callback:(void (^) (id response,NSError *error))callback{
    if(type == GET){
   
        [self GET:url parameters:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
            callback(responseObject,nil);
            
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            callback(nil, error);
        }];
    }else {
        [self POST:url parameters:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
          
            callback(responseObject,nil);
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            callback(nil,error);
        }];
    }
    
    
}


@end

File: 9.20/swift封装网络请求工具类/swift封装网络请求工具类/NetworkTool.swift
//
//  NetworkTool.swift
//  swift封装网络请求工具类
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//

import UIKit
import AFNetworking
enum RequestType: Int {
    case GET
    case POST
}
class NetworkTool: AFHTTPSessionManager {

    //创建一个单例接入点类方法创建
  
   
    static var sharedtool: NetworkTool = {
        let tool = NetworkTool()
        tool.responseSerializer.acceptableContentTypes?.insert("text/html")
        return tool
        
    }()
  

    
    //封装网络请求函数
    
    func requestType(type: RequestType,url: String ,parameters: Any?, callback: @escaping (Any?,Error?)->())  {
       
        if(type == .GET){
            get(url, parameters: nil, progress: nil, success: { (_, response) in
               
                callback(response,nil)
                }) { (_, error) in
                   
                    callback(nil,error)
            }
        }else{
            post(url, parameters: nil, progress: nil, success: { (_, resp) in
                callback(resp,nil)
                }, failure: { (_, error) in
                    callback(nil,error)
            })
        }
        
    }
    
    
    
}
